tensorboard/webapp/plugins/npmi/BUILD

```diff
@@ -56,8 +64,51 @@ py_test(
     srcs_version = "PY2AND3",
     deps = [
         ":summary",
+        ":metadata",
         "//tensorboard:expect_tensorflow_installed",
-        "//tensorboard/backend/event_processing:data_provider",
+        "//tensorboard/compat",
         "//tensorboard/util:test_util",
     ],
 )
+
+py_library(
+    name = "safe_encoder",
+    srcs = ["safe_encoder.py"],
+    srcs_version = "PY2AND3",
+    visibility = [
+        "//visibility:public",
+    ],
+    deps = [
+    ],
+)
+
+py_test(
+    name = "safe_encoder_test",
+    size = "small",
+    srcs = ["safe_encoder_test.py"],
+    srcs_version = "PY2AND3",
+    deps = [
+        ":safe_encoder",
+        "//tensorboard:expect_tensorflow_installed",
+        "//tensorboard/util:test_util",
+    ],
+)
+
+py_library(
+    name = "metadata",
+    srcs = ["metadata.py"],
+    srcs_version = "PY2AND3",
+    visibility = [
+        "//tensorboard:internal",
+    ],
+    deps = [
+        ":protos_all_py_pb2",
+        "//tensorboard/compat/proto:protos_all_py_pb2",
+        "//tensorboard/util:tb_logging",
```

Comment:
This dep shouldn't be needed here?

Reply:
Done in [30a3b091](https://github.com/tensorflow/tensorboard/pull/3799/commits/30a3b0910a88a82eaeac7eb99bd383f7f397e139)